### PR TITLE
Added HyAuth plugin

### PR DESCRIPTION
**Pull request by btaens**
Authorization plugin with virtual roles and hierarchy.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -247,6 +247,7 @@ Focus is on the specific task.
 - [OpAuth](https://github.com/wouter0100/cakephp-opauth) - Opauth provides a standardized method for PHP applications to interface with authentication providers.
 - [Tools:Passwordable](https://github.com/dereuromark/cakephp-tools) - Containing [Passwordable behavior](https://github.com/dereuromark/cakephp-tools/blob/cake3/docs/Behavior/Passwordable.md) for a DRY approach on password hashing.
 - [TinyAuth plugin](https://github.com/dereuromark/cakephp-tinyauth) - Role based (single/multi) authentication as very light-weight approach.
+- [HyAuth plugin](https://github.com/btaens/hy-auth) - A CakePHP plugin for hierarchical, role based, simple authorization.
```
Comment (dereuromark) [inline README.md]: Please respect alphabetical order :)
Also it would be good to have at least a basic test coverage (supported by travis CI).

**Comment by btaens** [comment]
Fixed the alphabetical order. I will write some tests soon.

**Comment by dereuromark** [comment]
Looks like the name could be confused with the existing HybridAuth one.
Maye I ask why it is named "Hy", when `hierarchical` does not contain a y? I would have thought to find a plugin for hierarchical auth stuff under sth more like `HieAuth` or better yet `HierAuth`. :)

**Comment by btaens** [comment]
'cause I'm dumb. HierAuth is a much better name. I'll rename the repo and reupload.

**Comment by btaens** [comment]
Alright, I've renamed the repo, and added testing and coverage.
Man, testing is hard.

**Comment by dereuromark** [comment]
98% coverage, wow, awesome
Yeah, it is, but its worth it! :)

**Pull request merged**